src/app/shared/services/dynamic-data/adapters/persistedMemory.ts

```diff
@@ -155,13 +155,37 @@ export class PersistedMemoryAdapter {
     this.persistStateToDB();
   }
 
+  public async deleteAll() {
+    this.state = {};
+    await this.persistStateToDB();
+  }
+
   /** Trigger persist handler. Requests will be debounced and notified when complete */
   public async persistStateToDB() {
     this.statePersist$.next("pending");
     // Wait for persist handler to notify of successful operation
     await firstValueFrom(this.statePersist$.pipe(filter((v) => v === "complete")));
   }
 
+  /** Return list of all flow name and flow type for each saved flow */
+  public async getAllFlowNamesAndTypes(): Promise<{ flow_type: string; flow_name: string }[]> {
+    const allCollections = Object.values(this.db.collections);
+
+    const allResults = await Promise.all(
+      allCollections.map(async (collection) => {
+        const docs: RxDocument[] = await collection.find().exec();
+        if (!docs || docs.length === 0) {
+          return null;
+        }
+        // Since each collection relates to a single flow, just take the first
+        const doc = docs[0].toMutableJSON();
+        const { flow_type, flow_name } = doc as any;
+        return { flow_type, flow_name };
+      })
+    );
+
+    return allResults.filter((result) => result !== null);
```

Comment:
nit(blocking): This seems like quite a specialised/niche method to be exposing publically from what is meant to be a fairly generalised service. In these types of situation I'd usually prefer to write a `hack` - style method in the service that just retrieves the private underlying db as required, e.g.  `dynamicDataService['writeCache']['db']`

For this specific case, I'm not even sure if it's required - struggling to follow fully but looks like you're trying to get a full list of tables used by dynamic data service so that you can then reset them one-by-one (excluding any that are internal)... I'm guessing it might be simpler to just wipe everything and and reload the app/page so that any internals can be re-initialised as required